app/controllers/reviewable_notes_controller.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+class ReviewableNotesController < ApplicationController
+  requires_login
+  before_action :ensure_staff
+  before_action :find_reviewable
+
+  def create
+    note = @reviewable.reviewable_notes.build(note_params)
+    note.user = current_user
+
+    if note.save
+      # Reload to ensure associations are loaded
+      note.reload
+      render json: ReviewableNoteSerializer.new(note, scope: guardian, root: false)
+    else
+      render json: { errors: note.errors.full_messages }, status: 422
+    end
+  end
+
+  def destroy
+    note = @reviewable.reviewable_notes.find(params[:id])
+
+    # Only allow the author or admin to delete notes
+    raise Discourse::InvalidAccess unless note.user == current_user || current_user.admin?
+
+    note.destroy
+    render json: success_json
+  end
+
+  private
+
+  def find_reviewable
+    @reviewable = Reviewable.find(params[:reviewable_id])
+  end
+
+  def note_params
+    params.require(:reviewable_note).permit(:content)
+  end
+
+  def ensure_staff
+    raise Discourse::InvalidAccess unless current_user&.staff?
+  end
```

Comment:
```suggestion
```

We can remove this since it is already defined in `ApplicationController`

https://github.com/discourse/discourse/blob/9613f657d0cd88a10296f4aec5f4c08e0ceee8a7/app/controllers/application_controller.rb#L727-L729